base/bucket_gocb.go

```diff
@@ -1294,26 +1299,61 @@ func (bucket CouchbaseBucketGoCB) Incr(k string, amt, def uint64, exp int) (uint
 }
 
 func (bucket CouchbaseBucketGoCB) GetDDoc(docname string, into interface{}) error {
-	LogPanic("Unimplemented method: GetDDoc()")
+
+	bucketManager, err := bucket.getBucketManager()
+	if err != nil {
+		return err
+	}
+
+	designDoc, err := bucketManager.GetDesignDocument(docname)
+
+	// TODO: this is of course sub-optimal to do an unnecessary marshal/unmarshal round trip.
+	// TODO: Looks like it might require changes to gocb or Sync Gateway code in order to avoid this round trip
```

Comment:
Fixed in 116c9da05386dcaedc28aa36a5de199cf21a4b40 -- it uses that approach, but will fallback to the marshal/unmarshal round trip if it's not an empty interface pointer.  Added to the test case to exercise this code path as well.